encrypted-media/util/utils.js

```diff
@@ -268,4 +268,27 @@ function test_exception(testCase /*...*/) {
     }
 }
 
+// Check that the events sequence (array of strings) matches the pattern (array of either strings, or
+// arrays of strings, with the latter representing a possibly repeating sub-sequence)
+function checkEventSequence(events,pattern) {
+    function th(i) { return i + (i < 4 ? ["th", "st", "nd", "rd"][i] : "th"); }
+    var i = 0, j=0, k=0;
```

Comment:
nit: More descriptive variable names, would be helpful, though also make the following code much longer.

Even commenting what the variables are would help. Using something like e, p, and pe _might_ be better. (Those are short for eventsIndex, patternIndex, and patternElementIndex.)